File: mogo_js/mapreduce.js
function wordMap() {
    var words = this.text.match(/\w+/g);

    if (words == null) {
        return;
    }

    for (var i = 0; i < words.length; i++) {
        emit(words[i], {count: 1});
    }
}

function wordReduce(key, values) {
    var total = 0;
    for (var i = 0; i < values.length; i++) {
        total += values[i].count;
    }

    return {count: total};
}

File: mogo_js/mapreduce.py
from pymongo import MongoClient
from pymongo.code import Code


connection = MongoClient('localhost', 50002)
db = connection.obama


# db.texts.remove()
with open("../mongo/obama.txt", "r") as f:
    lines = f.readlines()

[db.texts.insert({"text": line}) for line in lines]

with open("wordmap.js", "r") as f:
    map = Code(f.read())


with open("wordreduce.js", "r") as f:
    reduce = Code(f.read())


results = db.texts.map_reduce(map, reduce)

for result in results.find():
    print(result["id"], result["value"]["count"])
